File: _experimental/_elixir/numbers/lib/numbers_recursive.ex
defmodule NumbersRecursive do
  def fibonacci(0), do: 0
  def fibonacci(1), do: 1
  def fibonacci(n), do: fibonacci(n - 1) + fibonacci(n - 2)

  def factorial(0), do: 1
  def factorial(n), do: n * factorial(n - 1)

  def sum_numbers(0), do: 0
  def sum_numbers(n), do: n + sum_numbers(n - 1)
end

File: _experimental/_go/numbers/numbers_recursive/numbers_recursive.go
package numbers_recursive

func Fibonacci(n int) int {
	if n <= 0 {
		return 0
	} else if n == 1 {
		return 1
	}
	return Fibonacci(n-1) + Fibonacci(n-2)
}

func Factorial(n int) int {
	if n == 0 {
		return 1
	}
	return n * Factorial(n-1)
}

func SumNumbers(n int) int {
	if n <= 0 {
		return 0
	}
	return n + SumNumbers(n-1)
}
